lib/pat.c

```diff
@@ -6196,7 +6217,11 @@ grn_pat_wal_recover_add_entry(grn_ctx *ctx,
                               const char *tag,
                               const char *wal_error_tag)
 {
-  pat->header->curr_key = (uint32_t)(entry->key_offset);
+  if (pat_is_key_large(pat)) {
+    pat->header->curr_key_large = entry->key_offset;
+  } else {
+    pat->header->curr_key = (uint32_t)(entry->key_offset);
+  }
```

Comment:
Could we use `pat_update_curr_key` here too? And also, do we cast the updated value to `unit32_t`?

Reply:
cast is not necessary in this case.